x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/action/TransportPutEnrichPolicyAction.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.enrich.action;
+
+import org.elasticsearch.ResourceNotFoundException;
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.master.AcknowledgedResponse;
+import org.elasticsearch.action.support.master.TransportMasterNodeAction;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.block.ClusterBlockException;
+import org.elasticsearch.cluster.block.ClusterBlockLevel;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+import org.elasticsearch.xpack.core.enrich.action.PutEnrichPolicyAction;
+import org.elasticsearch.xpack.core.enrich.EnrichPolicy;
+import org.elasticsearch.xpack.enrich.EnrichStore;
+
+public class TransportPutEnrichPolicyAction extends TransportMasterNodeAction<PutEnrichPolicyAction.Request, AcknowledgedResponse> {
+
+    private final EnrichStore enrichStore;
+
+    @Inject
+    public TransportPutEnrichPolicyAction(TransportService transportService,
+                                          ClusterService clusterService,
+                                          ThreadPool threadPool,
+                                          ActionFilters actionFilters,
+                                          IndexNameExpressionResolver indexNameExpressionResolver,
+                                          EnrichStore enrichStore) {
+        super(PutEnrichPolicyAction.NAME, transportService, clusterService, threadPool, actionFilters,
+            PutEnrichPolicyAction.Request::new, indexNameExpressionResolver);
+        this.enrichStore = enrichStore;
+    }
+    @Override
+    protected String executor() {
+        return ThreadPool.Names.SAME;
+    }
+
+    @Override
+    protected AcknowledgedResponse newResponse() {
+        throw new UnsupportedOperationException("usage of Streamable is to be replaced by Writeable");
+    }
+
+    @Override
+    protected void masterOperation(PutEnrichPolicyAction.Request request, ClusterState state,
+                                   ActionListener<AcknowledgedResponse> listener) throws Exception {
+        // TODO license validation
+        putPolicy(request.getName(), request.getPolicy(), enrichStore, listener);
+    }
+
+    static void putPolicy(String name, EnrichPolicy policy, EnrichStore store, ActionListener<AcknowledgedResponse> listener) {
+        if (Strings.isNullOrEmpty(name)) {
```

Comment:
maybe add this if check to `PolicyStore#putStore(...)`?
and then replace `testGetEnricyPolicyAction_emptyChecks` test with a test in `EnrichStoreTests`

Reply:
and I think that this method can be merged with `masterOperation(...)`

Reply:
Also an `IllegalArgumentException` is properly better here. Specifying an empty name or no name is just not allowed when creating a policy. On the rest layer this is then translated to a 400 response code instead of a 404 response code.

Reply:
Given this, I think Im just going to remove the transport test. the REST and other tests exercise the transport layer fully, as it is a wrapper w/o validation, and since there is no license checks, its literally now a 1 line passthru to the `EnrichStore`.